Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,303 @@ function Test-DiscoveryService{
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting
+    )
+    # Clean out empty variables in default azureSettings.config
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+
+    $removeEmptyVariable = $azureIdentityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    $removeEmptyVariable.ParentNode.RemoveChild($removeEmptyVariable)
+
+    $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+}
+
+function Remove-XMLChildNodes {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string[]] $nodesToSearch,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [string] $childNodeAttributeSetting,
+        [string] $reverseOperator
+    )
+    # Clean out empty variables in default azureSettings.config
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+
+    foreach($nodeToSearch in $nodesToSearch)
+    {
+     $setAzureSettings = $azureIdentityScope.SelectSingleNode($nodeToSearch)
+
+     if($reverseOperator -eq "not equal")
+     {
+       $removeEmptyVariables = $setAzureSettings.ChildNodes | Where-Object {$_.$childNodeGetAttribute -ne $childNodeAttributeSetting}
+     }
+     else
+     {
+       $removeEmptyVariables = $setAzureSettings.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+     }
+
+     if($null -ne $removeEmptyVariables)
+     {
+      foreach($removeEmptyVariable in $setAzureSettings.variable){
+       $removeEmptyVariable.ParentNode.RemoveChild($removeEmptyVariable)
+      }
+     }
+    }
+    $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+}
+
+function Get-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting
+    )
+
+    $installationConfig = [xml](Get-Content $installConfigPath)
+    $identityScope = $installationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $existingChildNode = @{}
+
+    $existingChildNode = $identityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    if ($null -eq $existingChildNode)
+    {
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeSetting not found in install.config"
```

Comment:
Should not finding the node be concerning? Or is it ok to just continue, like not finding azureSecretName for example

Reply:
In the current use case, I would hope to find azureSecretName, so that would be an issue. The function is written though to be used more generically, so it could be used for other code.